### PR TITLE
📏 Add enumerated `ruleId` to all errors/warnings in the redux store

### PR DESCRIPTION
**Pull request by fwkoch**
These `ruleId`s categorize all myst build errors in a structured way, allowing further sorting/reporting/etc of errors downstream.

In addition to adding `ruleId`s, this also cleans up many places where we were still `session.log`-ing errors instead of correctly tracking them through our `addWarningForFile` plumbing which adds the error to the redux store.

### CODE REVIEW

File: packages/myst-cli/src/config.ts [hunk 1]
```diff
@@ -233,21 +249,23 @@ export function loadConfigAndValidateOrThrow(session: ISession, path: string) {
...
+  const conf = readConfig(session, vfile);
   const existingConf = selectors.selectLocalRawConfig(session.store.getState(), path);
   if (existingConf && JSON.stringify(conf) === JSON.stringify(existingConf)) {
     return;
   }
   session.store.dispatch(config.actions.receiveRawConfig({ path, file, ...conf }));
   const { site, project } = conf;
   if (site) {
-    validateSiteConfigAndSave(session, path, file, site);
+    validateSiteConfigAndSave(session, path, vfile, site);
```
Comment (rowanc1) [inline packages/myst-cli/src/config.ts:262]: Nice, it is good to see the full pass on error handling touch these again!!

File: packages/myst-common/src/ruleids.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+export enum RuleId {
+  // Frontmatter rules
+  validConfigStructure = 'valid-config-structure',
+  siteConfigExists = 'site-config-exists',
+  projectConfigExists = 'project-config-exists',
```
Comment (rowanc1) [inline packages/myst-common/src/ruleids.ts:5]: Very cool to see 80+ rules come into existence for this. Wow. :)

File: packages/myst-common/src/utils.ts [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
+import type { Position } from 'unist';
 import { map } from 'unist-util-map';
 import { customAlphabet } from 'nanoid';
 import type { Node, Parent, PhrasingContent } from 'myst-spec';
+import type { RuleId } from './ruleids.js';
 import type { AdmonitionKind, GenericNode, GenericParent } from './types.js';
 
 export type MessageInfo = {
-  node?: Node;
+  node?: Node | Position;
```
Comment (rowanc1) [inline packages/myst-common/src/utils.ts:11]: Nice, that is closer to `VFile` implementation.

**Review by rowanc1 — APPROVED** [review]
Really good to see the error functions getting some love. I am excited to see some of the documentation around the rule-ids in the future, but I think getting this in now is good to go, especially as other PRs are likely going to conflict with these changes!

File: packages/myst-common/src/ruleids.ts (hunk 2 above)
Comment (fwkoch) [reply packages/myst-common/src/ruleids.ts:5]: Hah, yeah, I'm not quite sure what the correct granularity is for these guys, but for now I think this works!

**Pull request merged**